cql3/expr/expression.cc

```diff
@@ -564,89 +564,11 @@ value_set intersection(value_set a, value_set b, const abstract_type* type) {
     return std::visit(intersection_visitor{type}, std::move(a), std::move(b));
 }
 
-bool is_satisfied_by(const binary_operator& opr, const evaluation_inputs& inputs) {
-    if (is<token>(opr.lhs)) {
-        // The RHS value was already used to ensure we fetch only rows in the specified
-        // token range. It is impossible for any fetched row not to match now.
-        // When token restrictions are present we forbid all other restrictions on partition key.
-        // This means that the partition range is defined solely by restrictions on token.
-        // When is_satisifed_by is used by filtering we can be sure that the token restrictions
-        // are fulfilled. In the future it will be possible to evaluate() a token,
-        // and we will be able to get rid of this risky if.
-        return true;
-    }
```

Comment:
Trying to `evaluate()` an expression with token inside will cause `on_internal_error`.
I wanted to do this change after `evaluate()` is fully implemented, but I guess it's possible to do it now as well.

It might not crash, maybe in the current code it's impossible to call this on an expression with `expr::token`, but I would feel safer if we did it later. There was some reason for this `if`, something might've caused a problem.

Reply:
I'd recommend doing #12906 before this change.
Without it the change might cause an unexpected `on_internal_error` on some edge cases that use `token()`.